src/python_testing/TC_CC_2_1.py

```diff
@@ -0,0 +1,397 @@
+            TestStep(
+                41, 'TH reads Primary6Y attribute from DUT and Verify that the DUT response contains an uint16 [Min:0 Max:0xfeff] if NumberOfPrimaries is 6 or more'),
+            TestStep(42, 'TH reads Primary6Intensity attribute from DUT and Verify that the DUT response contains an uint8 if NumberOfPrimaries is 6 or more'),
+            TestStep(43, 'TH reads from the DUT the (0x0030) WhitePointX attribute'),
+            TestStep(44, 'TH reads from the DUT the (0x0031) WhitePointY attribute'),
+            TestStep(45, 'TH reads from the DUT the (0x0032) ColorPointRX attribute'),
+            TestStep(46, 'TH reads from the DUT the (0x0033) ColorPointRY attribute'),
+            TestStep(47, 'TH reads from the DUT the (0x0034) ColorPointRIntensity attribute'),
+            TestStep(48, 'TH reads from the DUT the (0x0036) ColorPointGX attribute'),
+            TestStep(49, 'TH reads from the DUT the (0x0037) ColorPointGY attribute'),
+            TestStep(50, 'TH reads from the DUT the (0x0038) ColorPointGIntensity attribute'),
+            TestStep(51, 'TH reads from the DUT the (0x003a) ColorPointBX attribute'),
+            TestStep(52, 'TH reads from the DUT the (0x003b) ColorPointBY attribute'),
+            TestStep(53, 'TH reads from the DUT the (0x003c) ColorPointBIntensity attribute')
+        ]
+
+        return steps
+
+    async def verify_primary_index(self, primary_index: int) -> bool:
+        # Read all PrimaryN<X,Y,Intensity> attributes available in the cluster
+        instance_attribute_names = [attr for attr in self.attributes.__dict__.keys(
+        ) if not attr.startswith("__") and attr.startswith('Primary')]
+        # Attributes to search
+        primary_list = ["Primary{primary}X", "Primary{primary}Y", "Primary{primary}Intensity"]
+        # Validate the expected attribute is found.
+        for primary in primary_list:
+            attribute_str = primary.format(primary=primary_index)
+            if attribute_str not in instance_attribute_names or not await self.cluster_has_attribute(self.endpoint, getattr(self.attributes, attribute_str)):
+                return False
+        return True
+
+    async def _get_totalnumberofprimaries(self, primaries: int) -> int:
+        """Read the attributes which area Primaries<n>X,Y,Intensity attributes and return the total found and available in the current cluster.
+        Returns:
+            numberofprimaries(int) : Number of Primary attributes found.
+        """
+        numberofprimaries = 0
+        for pindex in range(1, primaries+1):
+            if await self.verify_primary_index(pindex):
+                numberofprimaries += 1
+            else:
+                # Dont break to count all
+                logger.warning(f"PrimaryN<X,Y,Intensity> with index {pindex} was not found in the cluster.")
```

Comment:
I don't understand why this would be a warning - it is expected to be not present.
Does the framework have a `logger.info` ? If so, could change to
```suggestion
                logger.info(f"PrimaryN<X,Y,Intensity> with index {pindex} was (correctly) not found in the cluster.")
```